core/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -182,6 +182,37 @@ class DeltaTable private[tables](
     executeDelete(None)
   }
 
+  /**
+   * Optimize data in the table that match the given `condition`.
+   *
+   * @param condition Boolean SQL expression
+   *
+   * @since 1.2.0
+   */
+  def optimize(condition: String): Unit = {
+    optimize(functions.expr(condition))
+  }
+
+  /**
+   * Optimize data in the table that match the given `condition`.
```

Comment:
Specify that condition should only contain the filters on partition columns, otherwise a [[AnalysisException]] is thrown. Same comment for the next API.

Reply:
Added to the comment